internal/v5/api_test.go

```diff
@@ -1479,6 +1471,74 @@ func (s *APISuite) TestMetaPermPutUnauthorized(c *gc.C) {
 	})
 }
 
+func (s *APISuite) TestMetaCanWriteNoAuth(c *gc.C) {
```

Comment:
Can you add a test for what happens if you do a can-write for a charm that is promulgated and in the stable channel, where the user is the owner but doesn't have Write permissions because promulgated. Although I have no idea what the right answer is.